app/forms/image-upload.tsx

```diff
@@ -640,3 +655,135 @@ export function CreateImageSideModalForm() {
     </SideModalForm>
   )
 }
+
+function BlockSizeNotice({
+  blockSize,
+  efiPartOffset,
+  isBootableCd,
+}: {
+  blockSize: number
+  efiPartOffset: number
+  isBootableCd: boolean
+}) {
+  if (efiPartOffset === -1 && !isBootableCd) return null
+
+  // TODO: make sure this logic is correct
+
+  if (blockSize === efiPartOffset) return null
```

Comment:
Oh, I just read this line. That should cover my concern.

Reply:
It would probably help to move this logic over into `validateImage` so you can see everything up front. When you say "follow the GPT suggestion", you mean 4096 is correct and we should avoid warning if possible? How can we detect hybrid GPT+MBR?

Reply:
oh sorry I meant hybrid GPT+ISO above. (hybrid GPT+MBR is a whole 'nother rabbit hole to fall down that we should ignore for this PR, and probably forever)

Reply:
What I mean is if a user uploads a hybrid GPT+ISO, and the GPT is at offset 512, then the suggested block device size should be 512. If the user has 2048 selected, we maybe don't bother saying otherwise, but if the user has 4096 selected, that definitely won't boot and we should suggest 512.